enos/modules/k8s_deploy_vault/main.tf

```diff
@@ -0,0 +1,161 @@
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    enos = {
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+
+    helm = {
+      source  = "hashicorp/helm"
+      version = "2.6.0"
+    }
+  }
+}
+
+locals {
+  helm_chart_settings = {
+    "server.ha.enabled"             = "true"
+    "server.ha.replicas"            = var.vault_instance_count
+    "server.ha.raft.enabled"        = "true"
+    "server.affinity"               = ""
+    "server.image.repository"       = var.image_repository
+    "server.image.tag"              = var.image_tag
+    "server.image.pullPolicy"       = "Never" # Forces local image use
+    "server.resources.requests.cpu" = "50m"
+    "server.limits.memory"          = "200m"
+    "server.limits.cpu"             = "200m"
+    "server.ha.raft.config"         = file("${abspath(path.module)}/raft-config.hcl")
+    "server.dataStorage.size"       = "100m"
+  }
+  all_helm_chart_settings = var.ent_license == null ? local.helm_chart_settings : merge(local.helm_chart_settings, {
+    "server.extraEnvironmentVars.VAULT_LICENSE" = var.ent_license
+  })
+
+  vault_address = "http://127.0.0.1:8200"
+
+  instance_indexes = [for idx in range(var.vault_instance_count) : tostring(idx)]
+
+  leader_idx    = local.instance_indexes[0]
+  followers_idx = toset(slice(local.instance_indexes, 1, 3))
+}
+
+resource "helm_release" "vault" {
+  name = "vault"
+
+  repository = "https://helm.releases.hashicorp.com"
+  chart      = "vault"
+
+  dynamic "set" {
+    for_each = local.all_helm_chart_settings
+
+    content {
+      name  = set.key
+      value = set.value
+    }
+  }
+}
+
+data "enos_kubernetes_pods" "vault_pods" {
+  kubeconfig_base64 = var.kubeconfig_base64
+  context_name      = var.context_name
+  namespace         = helm_release.vault.namespace
+  label_selectors = [
+    "app.kubernetes.io/name=vault",
+    "component=server"
+  ]
+
+  depends_on = [helm_release.vault]
+}
+
+resource "enos_vault_init" "leader" {
+  bin_path   = "/bin/vault"
+  vault_addr = local.vault_address
+
+  key_shares    = 5
+  key_threshold = 3
+
+  transport = {
+    kubernetes = {
+      kubeconfig_base64 = var.kubeconfig_base64
+      context_name      = var.context_name
+      pod               = data.enos_kubernetes_pods.vault_pods.pods[local.leader_idx].name
+      namespace         = data.enos_kubernetes_pods.vault_pods.pods[local.leader_idx].namespace
+    }
+  }
+}
+
+resource "enos_vault_unseal" "leader" {
+  bin_path    = "/bin/vault"
+  vault_addr  = local.vault_address
+  seal_type   = "shamir"
+  unseal_keys = slice(enos_vault_init.leader.unseal_keys_b64, 0, 3)
```

Comment:
Why can't we pass all of the contents from init? Are we trying to prove that we only need the threshold to unseal?

Reply:
Didn't try that and I'm not sure it'll work... since when you shamir unseal it says progress 1/threshold ... 2/threshold ... I'll try it out and see what happens.